### PR TITLE
Remove config.Complete from Config

### PR DESCRIPTION
**Pull request by ahmad-diaa**
**What type of PR is this?**
/kind cleanup

/assign @alculquicondor 

**What this PR does / why we need it**:
First step of simplifying Config processing in order to be able to eliminate it completely as both packages `cmd/kube-scheduler/app/options` and `cmd/kube-scheduler/app/config` are redundant.
**Which issue(s) this PR fixes**:

Part of #71861

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test

File: cmd/kube-scheduler/app/testing/testserver.go [hunk 1]
```diff
@@ -120,7 +121,16 @@ func StartTestServer(t Logger, customFlags []string) (result TestServer, err err
 
 	errCh := make(chan error)
 	go func(stopCh <-chan struct{}) {
-		if err := app.Run(config.Complete(), stopCh); err != nil {
+		if config.InsecureServing != nil {
```
Comment (alculquicondor) [inline cmd/kube-scheduler/app/testing/testserver.go]: This is introducing repeated code. What about moving the logic to `app.Run`?

Comment (ahmad-diaa) [reply cmd/kube-scheduler/app/testing/testserver.go]: Makes sense. Done!

**Review by draveness — COMMENTED** [review]
could you please squash the commits? thanks

File: cmd/kube-scheduler/app/server.go [hunk 2]
```diff
@@ -148,16 +147,24 @@ func runCommand(cmd *cobra.Command, args []string, opts *options.Options, regist
...
 		return fmt.Errorf("unable to register configz: %s", err)
 	}
 
-	return Run(cc, stopCh, registryOptions...)
+	return Run(c, stopCh, registryOptions...)
 }
 
 // Run executes the scheduler based on the given configuration. It only return on error or when stopCh is closed.
-func Run(cc schedulerserverconfig.CompletedConfig, stopCh <-chan struct{}, registryOptions ...Option) error {
+func Run(cc *schedulerserverconfig.Config, stopCh <-chan struct{}, registryOptions ...Option) error {
```
[the remark this answers is not in the document]
Comment (ahmad-diaa) [reply cmd/kube-scheduler/app/server.go]: Done

File: cmd/kube-scheduler/app/server.go [hunk 3]
```diff
@@ -148,16 +147,24 @@ func runCommand(cmd *cobra.Command, args []string, opts *options.Options, regist
...
+	if cc.InsecureServing != nil {
+		cc.InsecureServing.Name = "healthz"
+	}
+	if cc.InsecureMetricsServing != nil {
+		cc.InsecureMetricsServing.Name = "metrics"
+	}
+
+	apiserver.AuthorizeClientBearerToken(cc.LoopbackClientConfig, &cc.Authentication, &cc.Authorization)
```
[the remark this answers is not in the document]
Comment (ahmad-diaa) [reply cmd/kube-scheduler/app/server.go]: Done

**Comment by ahmad-diaa** [comment]
@ahg-g Does the on going discussion on https://github.com/kubernetes/kubernetes/issues/71861 about whether or not to remove the `Config` affect merging this PR?

**Comment by ahmad-diaa** [comment]
> I don't see a value to CompletedConfig struct, which is just a thing wrapper around Config.

I agree, it didn't make sense.

**Comment by sttts** [comment]
/hold

This moves kube-scheduler away from plans of component-base.

**Comment by sttts** [comment]
The addition of the config-option split might look unnecessary here in kube-scheduler, but it is the pattern we follow in all control plane components. The component-standard working group efforts are there to standardize our components. This PR works against it. 

cc @stealthybox

**Pull request closed without merging**